parla/dataflow.py

```diff
@@ -4,10 +4,10 @@
 and also eager-fashion automatic data movement.
 """
 
-from typing import Collection, Any
+from typing import List, Any
```

Comment:
Why `List` vs `Collection`? Is this just more specific?

Reply:
Yes. It's hinted by the linter that some `Collection` typed objects are using `+` operator which is not defined for the general `Collection`. It's a minor change but I think type hints should be as specific as possible.